lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -170,7 +170,7 @@ def connect(opts={})
 
     # Configure the HTTP client with the supplied parameter
     nclient.set_config(
-      'vhost' => opts['vhost'] || opts['rhost'] || self.vhost(),
```

Comment:
At this point `opts['rhost']` has already been resolved to the ip address so we never want that to be the vhost